specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2024-09-01/DataConnectors.json

```diff
@@ -0,0 +1,1696 @@
+        },
+        "grantType": {
+          "type": "string",
+          "description": "The grant type, usually will be 'authorization code'."
+        },
+        "tokenEndpoint": {
+          "type": "string",
+          "description": "The token endpoint. Defines the OAuth2 refresh token."
+        },
+        "tokenEndpointHeaders": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "The token endpoint headers."
+        },
+        "tokenEndpointQueryParameters": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "The token endpoint query parameters."
+        },
+        "authorizationEndpoint": {
+          "type": "string",
+          "description": "The authorization endpoint."
+        },
+        "authorizationEndpointHeaders": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "The authorization endpoint headers."
+        },
+        "authorizationEndpointQueryParameters": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "The authorization endpoint query parameters."
+        },
+        "isJwtBearerFlow": {
+          "type": "boolean",
+          "description": "A value indicating whether it's a JWT flow."
+        },
+        "accessTokenPrepend": {
+          "type": "string",
+          "description": "Access token prepend. Default is 'Bearer'."
+        }
+      },
+      "required": [
+        "clientSecret",
+        "clientId",
+        "grantType",
+        "tokenEndpoint"
+      ],
+      "type": "object",
+      "x-ms-discriminator-value": "OAuth2"
+    },
+    "OracleAuthModel": {
+      "description": "Model for API authentication for Oracle.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/CcpAuthConfig"
+        }
+      ],
+      "properties": {
+        "tenantId": {
+          "type": "string",
+          "description": "Oracle tenant ID"
+        },
+        "userId": {
+          "type": "string",
+          "description": "Oracle user ID"
+        },
+        "publicFingerprint": {
+          "type": "string",
+          "description": "Public Fingerprint"
+        },
+        "pemFile": {
+          "type": "string",
+          "description": "Content of the PRM file"
+        }
+      },
+      "type": "object",
+      "required": [
+        "tenantId",
+        "userId",
+        "publicFingerprint",
+        "pemFile"
+      ],
+      "x-ms-discriminator-value": "Oracle"
+    },
+    "SessionAuthModel": {
+      "description": "Model for API authentication with session cookie.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/CcpAuthConfig"
+        }
+      ],
+      "properties": {
+        "userName": {
```

Comment:
>userName [](http://example.com/codeflow?start=9&length=8)

ARM is trying to reduce instances of unschematized APIs to make it easier for consumers of those APIs to understand them and use them reliably, not to mention provide better support in the various SDKs.

Please make this an array or provide an explanation for why you need this.

Reply:
same. done in Preview Version.